api/envoy/extensions/filters/http/api_key_auth/v3/api_key_auth.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.api_key_auth.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/sensitive.proto";
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.api_key_auth.v3";
+option java_outer_classname = "ApiKeyAuthProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/api_key_auth/v3;api_key_authv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: APIKey Auth]
+// APIKey Auth :ref:`configuration overview <config_http_filters_api_key_auth>`.
+// [#extension: envoy.filters.http.api_key_auth]
+
+// API Key HTTP authentication.
+//
+// Example:
+//
+// .. code-block:: yaml
+//
+//    authentication_header: "X-API-KEY"
+//    keys:
+//      inline_string: |-
+//        clientID1:apiKey1
+//        clientID2:apiKey2
+//
+message APIKeyAuth {
+  // keys used to authenticate the client.
+  // It should be a map of clientID to apiKey. 
+  // The clientID serves solely for identification purposes and isn't used for authentication.
+  config.core.v3.DataSource keys = 1 [(udpa.annotations.sensitive) = true];
+
+  // The header name to fetch the key.
```

Comment:
nit: what happens when a header has multiple values? can we document the expected behavior?

Reply:
This is a good point.
I think we should return an error, but what does everyone think? cc @wbpcode @zhaohuabing 

I (only) checked kong's plugin implementation briefly, and they return an error ([ref](https://github.com/Kong/kong/blob/master/kong/plugins/key-auth/handler.lua#L168)).

Reply:
Our habit is only use the first header value. But I think it's OK to use the first value or treat it as error. :)